src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -232,43 +481,110 @@ public Authentication authenticate(Authentication authentication) throws Authent
+                        for (String userGroup : usersGroups) {                    	
+                        	// copy any of the user's permissions from the prior auth mgr to our new one
+                        	for (PermissionGroup pg : permissionGroups) {
+                        		for (Permission p : pg.getPermissions()) {
+                        			if (existingAuthMgr.hasPermission(userGroup, p)) {
+                        				newAuthMgr.add(p, userGroup);
+                        			}
+                        		}
+                        	}
+                        }
+                        
+                    	if (addInsteadOfCopy) {
+                        	if (LOGGER.isLoggable(Level.FINE))
+                        		LOGGER.fine(String.format("onSuccess: adding permissions to new user %s based on openshift roles %s", info.getName(), allowedRoles));
+                        	
+                        	// map OpenShift user based on role to Jenkins user with analogous permissions
+                        	//TODO can we minimally assume view role ?  assuming so while waiting for the build.oenshift.io.jenkins related changes we think are coming
+                        	// and see if a self-sar against the view role comes back and says allowed (currently returns allowed as false for the view role)
+                        	newAuthMgr.add(Hudson.READ, info.getName());
+                        	newAuthMgr.add(Item.READ, info.getName());
+                			if (allowedRoles.contains("edit") || allowedRoles.contains("admin")) {
+                				newAuthMgr.add(Item.BUILD, info.getName());
+                				newAuthMgr.add(Item.CONFIGURE, info.getName());
+                				newAuthMgr.add(Item.CREATE, info.getName());
+                				newAuthMgr.add(Item.DELETE, info.getName());
+                				newAuthMgr.add(Item.WORKSPACE, info.getName());
+                				newAuthMgr.add(SCM.TAG, info.getName());
+                			}
+                			if (allowedRoles.contains("admin")) {
+                				newAuthMgr.add(Computer.CONFIGURE, info.getName());
+                				newAuthMgr.add(Computer.DELETE, info.getName());
+                				newAuthMgr.add(Hudson.ADMINISTER, info.getName());
+                				newAuthMgr.add(Hudson.READ, info.getName());
+                				newAuthMgr.add(Run.DELETE, info.getName());
+                				newAuthMgr.add(Run.UPDATE, info.getName());
+                				newAuthMgr.add(View.CONFIGURE, info.getName());
+                				newAuthMgr.add(View.CREATE, info.getName());
+                				newAuthMgr.add(View.DELETE, info.getName());
+                			}
+                    	}
+            			
+            			Jenkins.getInstance().setAuthorizationStrategy(newAuthMgr);
```

Comment:
On Thu, Aug 25, 2016 at 2:48 PM, David Eads <notifications@github.com>
wrote:

> In src/main/java/org/openshift/jenkins/plugins/openshiftlogin/
> OpenShiftOAuth2SecurityRealm.java
> <https://github.com/openshift/jenkins-openshift-login-plugin/pull/2#discussion_r76302766>
> :
>
> > +                				newAuthMgr.add(SCM.TAG, info.getName());
> > +                			}
> > +                			if (allowedRoles.contains("admin")) {
> > +                				newAuthMgr.add(Computer.CONFIGURE, info.getName());
> > +                				newAuthMgr.add(Computer.DELETE, info.getName());
> > +                				newAuthMgr.add(Hudson.ADMINISTER, info.getName());
> > +                				newAuthMgr.add(Hudson.READ, info.getName());
> > +                				newAuthMgr.add(Run.DELETE, info.getName());
> > +                				newAuthMgr.add(Run.UPDATE, info.getName());
> > +                				newAuthMgr.add(View.CONFIGURE, info.getName());
> > +                				newAuthMgr.add(View.CREATE, info.getName());
> > +                				newAuthMgr.add(View.DELETE, info.getName());
> > +                			}
> > +                    	}
> > +            			
> > +            			Jenkins.getInstance().setAuthorizationStrategy(newAuthMgr);
>
> If a user loses access later on, does this clean up after them to revoke
> their access?
>
The next time the user logs in and creates a new session, yes, the new
permissions should be reflected.  They'll maintain the privileges they
possessed
when the logged into Jenkins for the life of that session currently.  In
theory we could add  background polling to augment things while the user's
session is active
but my initial thought is that is excessive (but let me know if you
disagree).

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/jenkins-openshift-login-plugin/pull/2/files/d32f29cf646040cbf44ea259d66012cdf6011e6c#r76302766>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADbadPqtbgwMoykYpY60fZJ2UkyTvtB2ks5qjeOUgaJpZM4Jfdh->
> .
>